app/helpers/patient_query_helper.rb

```diff
@@ -282,6 +282,18 @@ def advanced_filter(patients, filters, tz_offset)
         patients = advanced_filter_relative_date(patients, :symptom_onset, filter, tz_diff, :date)
       when 'continous-exposure'
         patients = patients.where(continuous_exposure: filter[:value].present? ? true : [nil, false])
+      when 'close-contact-with-known-case-id'
+        if filter[:value].blank?
+          patients = patients.where(contact_of_known_case_id: [nil, ''])
+        else
+          value_string = filter[:value].split(/\s*,\s*/).join('|')
+          case filter[:additionalFilterOption]
+          when 'Exact Match'
+            patients = patients.where('contact_of_known_case_id REGEXP ?', "(^|,\s?)(#{value_string})(\s?,|$)")
+          when 'Contains'
+            patients = patients.where('contact_of_known_case_id REGEXP ?', value_string.to_s)
```

Comment:
Why do you need `to_s` here?

Also, part of the reason I didn't actually suggest REGEXP for the "Contains" case is because LIKE is much more performant than REGEXP, so when we can use LIKE (and have it be faster), we should. I think for this case it may be faster to just build a LIKE query with ORs, based on the benchmarks. The "Exact Match" case above I think just lends itself more to a REGEXP for sure.

![Screen Shot 2021-02-26 at 11 58 07 AM](https://user-images.githubusercontent.com/11698457/109330807-16b74080-782a-11eb-910d-fd9e972387b6.png)

That being said, this may not be run that often and I'm not sure it'd be _that_ much faster in this case - I'd need to run on many more monitorees locally. @holmesie What are your thoughts?

Reply:
wasn't the problem with using `LIKE` getting results of like `243` when you're searching for `24` or `43`?

Reply:
We were using LIKE before for the "Contains" case. The problem you mentioned would be for the "Exact Match" case I think

Reply:
@kylietmo the `to_s` was flagged as necessary in rubocop, I forget exactly why

I was not able to get the `LIKE` to work like that (I tried for a bit).  Using the regexp is cleaner anyway so I'd rather keep it this way

Reply:
It's definitely do-able to get the LIKE to work with this case. Agreed this is cleaner, but we need to make sure we're making smart trade-offs when it comes to cleaner vs more performant code. I think this is probably fine, but I'd like to know what @holmesie thinks before resolving.

Reply:
I would agree that performance is more important here than making it look cleaner

Reply:
Discussed with Adam, and we think keeping the REGEX is fine for now. We can keep an eye on these queries as well and revisit in the future if need be.